app/src/main/java/de/westnordost/streetcomplete/quests/crossing_island/AddCrossingIsland.kt

```diff
@@ -0,0 +1,58 @@
+package de.westnordost.streetcomplete.quests.crossing_island
+
+import de.westnordost.osmapi.map.data.BoundingBox
+import de.westnordost.osmapi.map.data.Element
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.elementgeometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.mapdata.OverpassMapDataAndGeometryApi
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmElementQuestType
+import de.westnordost.streetcomplete.data.tagfilters.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.tagfilters.toGlobalOverpassBBox
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment
+
+class AddCrossingIsland(private val overpass: OverpassMapDataAndGeometryApi)
+    : OsmElementQuestType<Boolean> {
+
+    override val commitMessage = "Add whether pedestrian crossing has an island"
+    override val wikiLink = "Key:crossing:island"
+    override val icon = R.drawable.ic_quest_pedestrian_crossing_island
+
+    override fun getTitle(tags: Map<String, String>) =
+        R.string.quest_pedestrian_crossing_island
+
+    override fun isApplicableTo(element: Element): Boolean? = null
+
+    override fun download(bbox: BoundingBox, handler: (element: Element, geometry: ElementGeometry?) -> Unit): Boolean {
+        return overpass.query(getOverpassQuery(bbox), handler)
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox): String {
+        val accessFilter = "\"access\"~\"^(private|no)\$\""
+        val onewayFilter = "\"oneway\"!=\"no\""
+        val pathFilter = "\"highway\"~\"^(path|footway|cycleway|pedestrian)\$\""
```

Comment:
I don't think this increases readability, especially since you have to escape all the `"` there as well.

Reply:
Also, why did you escape the dollar? I think this is a mistake

Reply:
Well. It works with escaping and it threw an exception (due to an error vom overpass) without them.
I'll just reintrage them in the OQL string.

Reply:
I also don't think the variables in the current form improve readability, but you can at least use triple quotes (for a string literal in kotlin) there too (like in the return) to get rid of the need to escape the quotes.